web/app/scripts/export/export-controller.js

```diff
@@ -0,0 +1,98 @@
+     * Exports are tied to ctl.tilekey, which comes via the directive from MapController.
+     * Changing filters changes tilekey which resets the download.
+     */
+
+    /* ngInject */
+    function ExportController($rootScope, $scope, $interval, Exports, InitialState) {
+        var ctl = this;
+        var pollingInterval;
+        var POLLING_INTERVAL_MS = 1500;
+        var MAX_POLLING_TIME_S = 100;
+
+        InitialState.ready().then(initialize);
+
+        function initialize() {
+            ctl.isOpen = false;
+            ctl.pending = false;
+            ctl.downloadURL = null;
+            ctl.error = null;
+
+            ctl.toggle = toggle;
+            ctl.exportCSV = exportCSV;
+        }
+
+        function toggle() {
+            ctl.isOpen = !ctl.isOpen;
+            if (ctl.isOpen) {
+                $rootScope.$broadcast('driver.export.open');
+            }
+        }
+
+        function exportCSV() {
+            ctl.error = null;
+            Exports.create({ tilekey: ctl.tilekey },
+                function (result) {
+                    pollForDownload(result.taskid);
+                }, function () {
+                    ctl.error = 'Error initializing export.';
+                }
+            );
+        }
+
+        function pollForDownload(taskID) {
+            ctl.pending = true;
+            pollingInterval = $interval(function () {
+                    Exports.get({ id: taskID }).$promise.then(function (response) {
+                        switch (response.status) {
+                            case 'PENDING':
+                            case 'STARTED':
+                                break;
+                            case 'FAILURE':
+                                ctl.error = response.error;
+                                cancelPolling();
+                                break;
+                            case 'SUCCESS':
+                                ctl.downloadURL = response.result;
+                                cancelPolling();
+                                break;
+                        }
+                    });
+                },
+                POLLING_INTERVAL_MS,
+                MAX_POLLING_TIME_S * 1000 / POLLING_INTERVAL_MS
+            );
+            // The interval's promise resolves if it hits the limit without being cancelled
+            pollingInterval.then(function () {
+                cancelPolling();
+                ctl.error = 'Export request timed out.';
+            });
+        }
+
+        function cancelPolling() {
+            ctl.pending = false;
+            $interval.cancel(pollingInterval);
+        }
+
+        // Clears any in-progress or completed download, so they can do a new one if they want.
+        function reset() {
+            cancelPolling();
+            ctl.error = null;
+            ctl.downloadURL = null;
+        }
+
+        $scope.$watch(function () { return ctl.tilekey; }, reset);
```

Comment:
Yeah, I see your point.  I can make the download link show up below the export link rather than replacing it.  Question: do you think it's worth watching the filter params and only showing the export link when the filters don't match the most recent download, or should we just let them export whenever there's not one already pending, even if the answer will be the same?

I guess technically in production the answer wouldn't necessarily be the same, since other people could be adding or modifying records.  Probably the latter, then.